src/sentry/integrations/github_enterprise/integration.py

```diff
@@ -225,18 +226,22 @@ def build_integration(self, state):
             identity['access_token'],
             state['installation_id'])
 
+        domain = urlparse(installation['account']['html_url']).netloc
         return {
             'name': installation['account']['login'],
-            'external_id': installation['id'],
+            # installation id is not enough to be unique for self-hosted GH
+            'external_id': '{}:{}'.format(domain, installation['id']),
             # GitHub identity is associated directly to the application, *not*
             # to the installation itself.
-            'idp_external_id': installation['app_id'],
+            # app id is not enough to be unique for self-hosted GH
+            'idp_external_id': '{}:{}'.format(domain, installation['app_id']),
             'metadata': {
                 # The access token will be populated upon API usage
                 'access_token': None,
                 'expires_at': None,
                 'icon': installation['account']['avatar_url'],
-                'domain_name': installation['account']['html_url'].replace('https://', '').split("/")[0],
+                'domain_name': installation['account']['html_url'],
+                'installation_id': installation['id'],
```

Comment:
We might end up also having to store app ID in here too, but I'll let you deal with that in the other PR

Reply:
it should be `metadata['installation']['id']` I thought?

Reply:
You're right!